dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchVectorStoreRecordCollection.cs

```diff
@@ -22,7 +22,7 @@ namespace Microsoft.SemanticKernel.Connectors.AzureAISearch;
 /// </summary>
 /// <typeparam name="TRecord">The data model to use for adding, updating and retrieving data from storage.</typeparam>
 #pragma warning disable CA1711 // Identifiers should not have incorrect suffix
-public sealed class AzureAISearchVectorStoreRecordCollection<TRecord> : IVectorStoreRecordCollection<string, TRecord>, IVectorSearch<TRecord>
+public sealed class AzureAISearchVectorStoreRecordCollection<TRecord> : IVectorStoreRecordCollection<string, TRecord>, IVectorizableTextSearch<TRecord>
```

Comment:
Is it necessary to specify implementation of the `IVectorStoreRecordCollection` directly if it's already specified indirectly via `IVectorStoreRecordCollection` interface which inherits from the `IVectorStoreRecordCollection` interface?

Reply:
I suppose you mean is it necessary to specify `IVectorizedSearch` if it already inherits from `IVectorStoreRecordCollection`?  Note that there are two interfaces in play here.  `IVectorizedSearch` and `IVectorizableTextSearch`.  The first is a base for `IVectorStoreRecordCollection`, the second is not.  This was one of the main issues in the review, in that the options are either have the VectorizableText bit on the main interface, and throw were not supported, or have it separate.